pkg/backend/httpstate/client/client.go

```diff
@@ -415,6 +424,46 @@ func (pc *Client) StartUpdate(ctx context.Context, update UpdateIdentifier,
 	return resp.Version, resp.Token, nil
 }
 
+// PublishPolicyPack publishes a `PolicyPack` to the Pulumi service.
+func (pc *Client) PublishPolicyPack(ctx context.Context, orgName string,
+	analyzerInfo plugin.AnalyzerInfo, dirArchive *bytes.Buffer) error {
+
+	//
+	// Step 1 of 2: Send POST containing policy metadata to service. This begins process of creating
+	// publishing the PolicyPack.
+	//
+
+	req := apitype.CreatePolicyPackRequest{
+		Name:        analyzerInfo.Name,
+		DisplayName: analyzerInfo.DisplayName,
+		Policies:    analyzerInfo.Policies,
+	}
+
+	var resp apitype.CreatePolicyPackResponse
+	err := pc.restCall(ctx, "POST", publishPolicyPackPath(orgName), nil, req, &resp)
+	if err != nil {
+		return errors.Wrapf(err, "HTTP POST to publish policy pack failed")
+	}
+
+	//
+	// Step 2 or 2: Upload the compressed PolicyPack directory to the presigned S3 URL. The
+	// PolicyPack is now published.
+	//
+
+	client := &http.Client{}
+	putS3Req, err := http.NewRequest(http.MethodPut, resp.UploadURI, dirArchive)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to upload compressed PolicyPack")
+	}
+
+	_, err = client.Do(putS3Req)
```

Comment:
Could this just be `http.DefaultClient` instead (and then not need to create the `client` var on L453?